core/shared/src/main/scala/fs2/Pull.scala

```diff
@@ -33,12 +33,87 @@ import fs2.internal._
 import Resource.ExitCase
 import Pull._
 
-/** A `p: Pull[F,O,R]` reads values from one or more streams, returns a
-  * result of type `R`, and produces a `Stream[F,O]` when calling `p.stream`.
+/** A pure abstraction representing the intention to perform a sequence of actions
+  * in an effect type `F`, which emits any number of values of an output type `O`.
+  * This sequence may end up in several ways:
+  * a. it may not terminate and keep unrolling new items;
+  * b. it may be aborted by a `Throwable` error;
+  * c. be interrupted by a signal, which may include an error,
+  * d. terminate successfully and returning one single carry-over result of type `R`.
   *
-  * Any resources acquired by `p` are freed following the call to `stream`.
+  * Much like the `IO` values fromo the `cats-effect` library,
+  * `Pull` values are pure, immutable values. Thus, they preserve referential
+  * transparency and are usable in functional programming.
+  * A `Pull` is a data structure that represents just a description
+  * of a sequence of potentially side effectful computation.
   *
-  * Laws:
+  * A pull emits the output values not one-at-a-time, but one-chunk at a time.
```

Comment:
Suggest: "For efficiency, output values are processed in chunks. A chunk is an immutable sequence with O(1) indexed lookup. A `Pull[F, Byte, R]` operates internally on `Chunk[Byte]` values, which can wrap unboxed byte arrays -- avoiding significant boxing/unboxing costs. The `Pull` API provides mechanisms for working at both the chunk level and the individual element level. Generally, working at the chunk level will result in better performance but at the cost of more complex implementations."

Reply:
Saying that it is for efficiency suggests that it is a merely operational concern, to be optimised later. 

However, as has been pointed elsewhere, Chunks are also boundaries of laziness for interruptions and terminations, so they are part of the observable "value-level", correctness.

Nevertheless, it is a paragraph to add.